Snowflake.Data/Core/SFDataConverter.cs

```diff
@@ -184,30 +196,33 @@ private static DateTimeOffset ConvertToDateTimeOffset(UTF8Buffer srcVal, SFDataT
             100000000 
         };
 
-        private static Tuple<long, long> ExtractTimestamp(UTF8Buffer srcVal)
+        private static long GetTicksFromSecondAndNanosecond(UTF8Buffer srcVal)

```

Comment:
Maybe add a comment on when this is method is used. I guess this is used where caller only needs precision of microseconds ? since it looks like you divide 100 at the end of this function?

Reply:
.Net DateTime and DateTimeOffset don't support nanoseconds precision. The best you can get are Ticks.
A single tick represents one hundred nanoseconds. This is why it's divided by 100 at the end. If you look at the original code, you'll see that this was done for each data in ConvertToDateTime before. I simply moved it to a common place to avoid duplication and confusion.
This method does return the number of ticks for the given seconds and nanoseconds value. I tried to get a meaningful name for it, looks like I failed :D I'll add a comment and submit the PR.